draft-ietf-dnsop-svcb-https.md

```diff
@@ -1228,6 +1228,68 @@ HTTPS RRs, all connections will be upgraded to HTTPS, and clients will
 
+### Multi-CDN {#multicdn}
+
+The HTTPS RR is intended to support HTTPS services operated by
+multiple independent entities, such as different Content Delivery
+Networks (CDNs) or different hosting providers.  This includes
+the case where a service is migrated from one operator to another,
+as well as the case where the service is multiplexed between
+multiple operators for performance, redundancy, etc.
+
+This example shows such a configuration, with www.customer.example
+having different DNS responses to different queries, either over time
+or due to logic within the authoritative DNS server:
+
+     $ORIGIN customer.example.  ; A Multi-CDN customer domain
+     ; This zone contains/returns different CNAME records
+     ; at different points-in-time.  The RRset for "www" can
+     ; only ever contain a single CNAME.  Sometimes the zone has:
+     www 900 IN CNAME cdn1.svc1.example.      ; For some subset of DNS responses
+     ; and other times it contains:
+     www 900 IN CNAME customer.svc2.example.  ; For the remaining DNS responses
+
+     ; The apex is also aliased to www to match its configuration
+     @     7200 IN HTTPS 0 www
+     ; Non-HTTPS-aware clients use non-CDN IPs
+                   A    203.0.113.82
+                   AAAA 2001:db8:203::2
+
+     ; Resolutions following the cdn1.svc1.example path use these records.
+     ; This CDN uses a different alternative service for HTTP/3.
+     $ORIGIN svc1.example.  ; domain for CDN 1
+     cdn1     1800 IN HTTPS 1 h3pool alpn=h3 echconfig="123..."
+                      HTTPS 2 . alpn=h2 echconfig="123..."
+                      A    192.0.2.2
+                      AAAA 2001:db8:192::4
+     h3pool 300 IN A 192.0.2.3
+                AAAA 2001:db8:192:7::3
+
+     ; Resolutions following the customer.svc2.example path use these records.
+     ; Note that this CDN only supports HTTP/2.
+     $ORIGIN svc2.example. ; domain operated by CDN 2     
+     customer 300 IN HTTPS 1 . alpn=h2 echconfig="xyz..."
+               60 IN A    198.51.100.2
+                     A    198.51.100.3
+                     A    198.51.100.4
+                     AAAA 2001:db8:198::7
+                     AAAA 2001:db8:198::12
+
+Note that in the above example, the different CDNs have different
+echconfig and different capabilities, but clients will use HTTPS RRs
+as a bound-together unit.
+
+Domain owners should be cautious when using a multi-CDN configuration, as it
+introduces a number of complexities highlighted by this example:
+
+* If CDN 1 supports ECH, and CDN 2 does not, the client is vulnerable to ECH
+  downgrade by a network adversary who forces clients to get CDN 2 records.
+
+* Aliasing the apex to its subdomain simplifies the zone file but likely
+  increases resolution latency, especially when using a non-HTTPS-aware
+  recursive resolver.  An alternative would be to alias the zone
+  apex directly to a name managed by a CDN.
+
```

Comment:
Probably also worth mentioning that the A/AAAA/HTTPS resolutions can follow different CNAMEs to different CDNs:
- The client will properly handle the HTTPS record, but find the reference to a hostname it hasn't started resolving yet; this could be fixed by including the IP hints.
- If only one CDN publishes HTTPS records, sometimes the client will receive NXDATA for the HTTPS query, but A/AAAA results for an ECH-capable CDN.

Reply:
Added notes about this